packages/engine/src/interaction/systems/InteractableSystem.ts

```diff
@@ -0,0 +1,325 @@
+  }
+}
+
+export const InteractableSystem = defineSystem({
+  uuid: 'ee.engine.InteractableSystem',
+  insert: { before: TransformSystem },
+  execute
+})
+
+const executeInput = () => {
+  if (getState(EngineState).isEditor) return
+
+  const inputPointerEntity = InputPointerComponent.getPointerForCanvas(Engine.instance.viewerEntity)
+  if (!inputPointerEntity) return
+
+  const buttons = InputSourceComponent.getMergedButtons()
+
+  const nonCapturedInputSource = InputSourceComponent.nonCapturedInputSources()
+  for (const entity of nonCapturedInputSource) {
+    const inputSource = getComponent(entity, InputSourceComponent)
+    if (buttons.KeyE?.down || inputSource.buttons[XRStandardGamepadButton.Trigger]?.down) {
+      interactWithClosestInteractable()
+    }
+  }
+
+  for (const entity of hoverInputInteractablesQuery()) {
+    const capturingEntity = getState(InputState).capturingEntity
+    const inputComponent = getComponent(entity, InputComponent)
+    const inputSourceEntity = inputComponent?.inputSources[0]
+
+    if (inputSourceEntity) {
+      const inputSource = getOptionalComponent(inputSourceEntity, InputSourceComponent)
+      if (capturingEntity !== UndefinedEntity) {
+        // return
+
+        const clickButtons = inputSource?.buttons
+        clicking = !!clickButtons //clicking on our boundingbox this frame
+
+        //TODO firing play on video each click, but for some reason only plays first time
+        //TODO refactor this, changing the execute timing is the only thing that makes this logic work, otherwise timings are different
+        //between PrimaryClick.up and capturingEntity being undefined or not
+        if (clicking && clickButtons) {
+          if (
+            clickButtons.PrimaryClick?.touched /*&& clickButtons.PrimaryClick.up*/ ||
+            clickButtons[XRStandardGamepadButton.Trigger]?.down
+          ) {
+            clickInteract(entity)
+          }
+        }
+      }
+    }
+
+    if (clicking && !inputSourceEntity && capturingEntity === UndefinedEntity) {
+      clicking = false
+    }
+  }
+}
+//TODO only activate the one interactable closest to the camera center and within range or hovered
+//TODO look into the design language (opacity, font size, etc) to differentiate between UI on and targeted for activation
+
+let clicking = false
+
+const clickInteract = (entity: Entity) => {
+  const interactable = getComponent(entity, InteractableComponent)
+  for (const callback of interactable.callbacks) {
+    if (callback.target && !UUIDComponent.getEntityByUUID(callback.target)) continue
+    const targetEntity = callback.target ? UUIDComponent.getEntityByUUID(callback.target) : entity
+    if (targetEntity && callback.callbackID) {
+      const callbacks = getOptionalComponent(targetEntity, CallbackComponent)
+      if (!callbacks) continue
+      callbacks.get(callback.callbackID)?.(entity, targetEntity)
+    }
+  }
+}
+
+const interactWithClosestInteractable = () => {
```

Comment:
Just a note for the next PR; This should be `interactWithFocussedInteractable(inputSource:Entity)`. The heuristic should be something like this: 

- Find the interactable that is closest to input source's target ray in the viewer's viewing frustum
  - This could be the closest interactable that intersects with that ray, or;
  - The closest interactable to that ray
- Run the interact callback on that interactable

Reply:
will create an issue, thanks